In [ ]:
%%bash
clear # used to clear the screen
# or use ctrl+L

In [ ]:
%%bash
exit
# or ctrl+D to come out of shell

In [ ]:
%%bash
ls -a
# -a option gives hidden files also

.
..
Linux_shell_course.ipynb
SC_Week0_Unit0.ipynb
SC_Week0_Unit1.ipynb
SC_Week0_Unit2.ipynb
SC_Week0_Unit4.ipynb
SC_Week0_Unit5.ipynb


In [ ]:
%%bash
ls -l
# -l option gives long format

total 270
-rw-r--r-- 0 kavis 197609  11069 Oct  6 11:21 Linux_shell_course.ipynb
-rw-r--r-- 0 kavis 197609   8173 Oct  5 19:51 SC_Week0_Unit0.ipynb
-rw-r--r-- 0 kavis 197609 230730 Oct  5 19:51 SC_Week0_Unit1.ipynb
-rw-r--r-- 0 kavis 197609  16186 Oct  5 20:28 SC_Week0_Unit2.ipynb
-rw-r--r-- 0 kavis 197609   3531 Oct  5 19:52 SC_Week0_Unit4.ipynb
-rw-r--r-- 0 kavis 197609   4159 Oct  5 19:51 SC_Week0_Unit5.ipynb


In [ ]:
%%bash
ls --help
#summary of how to use the command

Usage: ls [OPTION]... [FILE]...
List information about the FILEs (the current directory by default).
Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.

Mandatory arguments to long options are mandatory for short options too.
  -a, --all                  do not ignore entries starting with .
  -A, --almost-all           do not list implied . and ..
      --author               with -l, print the author of each file
  -b, --escape               print C-style escapes for nongraphic characters
      --block-size=SIZE      with -l, scale sizes by SIZE when printing them;
                               e.g., '--block-size=M'; see SIZE format below
  -B, --ignore-backups       do not list implied entries ending with ~
  -c                         with -lt: sort by, and show, ctime (time of last
                               modification of file status information);
                               with -l: show ctime and sort by name;
                               othe

In [ ]:
%%bash
man ls
clear
# doesn't work with vs code
# manual page for the command, which provides detailed documentation.

LS(1)                     User Commands                    LS(1)

NAME
       ls - list directory contents

SYNOPSIS
       ls [OPTION]... [FILE]...

DESCRIPTION
       List  information  about the FILEs (the current directory
       by default).  Sort entries alphabetically if none of -cf‐
       tuvSUX nor --sort is specified.

       Mandatory arguments to long  options  are  mandatory  for
       short options too.

       -a, --all
              do not ignore entries starting with .

       -A, --almost-all
              do not list implied . and ..

       --author
              with -l, print the author of each file

       -b, --escape
              print C-style escapes for nongraphic characters

       --block-size=SIZE
              with  -l,  scale sizes by SIZE when printing them;
              e.g., '--block-size=M'; see SIZE format below

       -B, --ignore-backups
              do not list implied entries ending with ~

       -c     with -lt: sort by, and show, ctime 

Filesystem Hierarchy Standard followed by all linux systems
FHS 3.0 released on June 03, 2015
Available at
https://refspecs.linuxfoundation.org/fhs.shtml

In [ ]:
%%bash
cd ..
##change directory to parent directory of current directory (parent directory means just one level up)
pwd

cd . 
## single dot stands for present directory so, no change
# cd . means stay in the current directory.
pwd


cd mnt/c/Users/kavis/Upskilling/'Github repository'/Rep1
pwd
#this path is relative to current directory and it doesn't actually exist

cd /mnt/c/Users/kavis/Upskilling/'Github repository'/Rep1
pwd
#absolute path

# use the command cd ~ or cd $HOME
cd ~
# The home directory is the starting directory for a user
pwd


/mnt/c/Users/kavis/Upskilling/Github repository/Rep1
/mnt/c/Users/kavis/Upskilling/Github repository/Rep1


bash: line 11: cd: mnt/c/Users/kavis/Upskilling/Github repository/Rep1: No such file or directory


/mnt/c/Users/kavis/Upskilling/Github repository/Rep1
/mnt/c/Users/kavis/Upskilling/Github repository/Rep1
/home/kavis


In [ ]:
%%bash
pwd

cd     # cd without arguement takes to the home directory
pwd

cd /    #root directory  (/) is the highest level in the filesystem hierarchy 
pwd
ls -a

#When you list all items in a directory using ls -a, you will see ..:
#   . represents the current directory.
#  .. always represents the parent directory, even if it’s a reference to the same directory (as is the case in the root directory).

cd /
pwd
cd ..
pwd

/g/My Drive/Upskilling/Github repository/Rep1/Linux_shell
/c/Users/kavis
/
.
..
LICENSE.txt
ReleaseNotes.html
bin
cmd
dev
etc
git-bash.exe
git-cmd.exe
mingw64
proc
tmp
unins001.dat
unins001.exe
unins001.msg
usr
/
/


In [ ]:
%%bash
pwd
cd /c
ls -l

/g/My Drive/Upskilling/Github repository/Rep1/Linux_shell
total 10793673
drwxr-xr-x 1 kavis 197609          0 Oct 20  2022 $Recycle.Bin
-rw-r--r-- 1 kavis 197609         40 Oct 21  2022 2CA0AE717CAA
lrwxrwxrwx 1 kavis 197609          8 Oct 11  2022 Documents and Settings -> /c/Users
drwxr-xr-x 1 kavis 197609          0 Jul 30  2022 Drivers
-rw-r--r-- 1 kavis 197609      12288 Jul 12 09:14 DumpStack.log
-rw-r--r-- 1 kavis 197609      12288 Oct  5 18:18 DumpStack.log.tmp
drwxr-xr-x 1 kavis 197609          0 May  4 13:43 OneDriveTemp
drwxr-xr-x 1 kavis 197609          0 May  7  2022 PerfLogs
drwxr-xr-x 1 kavis 197609          0 Oct  6 09:13 Program Files
drwxr-xr-x 1 kavis 197609          0 Jun 11 08:47 Program Files (x86)
drwxr-xr-x 1 kavis 197609          0 Sep 19 22:02 ProgramData
drwxr-xr-x 1 kavis 197609          0 Aug 10  2023 Recovery
-rw-r--r-- 1 kavis 197609       1505 Dec 19  2023 Setup.log
drwxr-xr-x 1 kavis 197609          0 Oct  6 09:12 System Volume Information
drwxr-xr-x 1 

#### Structure of Each Line:
1. **Permissions**: (`drwxr-xr-x`)
   - This shows the file permissions for the owner, group, and others.
   - first letter -  `d` indicates a **directory** (if it were a file, it would be `-`).
                     'l' indicates symbolic link (shortcut)
                     '-' indicates file
                     'c' character device
                     'b' block devices (hard discs)
   - next three for owner - `rwx` means **read, write, and execute** permissions for the owner.
   - next three for group - `r-x` means **read and execute** permissions for the group.
   - next three for others- `r-x` means **read and execute** permissions for others.

2. **Links**: (`1`)
   - The number of links or directories inside this directory.

3. **Owner**: (`kavis`)
   - The username of the owner of the file or directory.

4. **Group**: (`197609`)
   - The group ID associated with the file.

5. **Size**: (`0`, `40`, `12288`, etc.)
   - The size of the file in bytes. For directories, it’s usually `0` since it doesn’t directly reflect the size of contents.

6. **Timestamp**: (`Oct 6 14:21`)
   - The date and time when the file or directory was last modified.

7. **File/Directory Name**: (`$Recycle.Bin`, `Documents and Settings`, etc.)
   - The name of the file or directory.

#### Listing Explanation:

**`Documents and Settings -> /c/Users`**:
   - This is a **symbolic link** (shortcut) that points to `/c/Users`. It means that trying to access `Documents and Settings` will actually redirect you to the `/c/Users` directory.

**`Drivers`**:
   - Likely a folder containing drivers for the system. Drivers are software that help the operating system communicate with hardware devices.

**`Program Files`** and **`Program Files (x86)`**:
   - Standard directories in Windows for installing 64-bit and 32-bit applications, respectively.

**`ProgramData`**:
   - A directory used to store application data that is not specific to a user. It’s shared among all users of the system.

**`Users`**:
    - This is where user profiles are stored on Windows, such as `Desktop`, `Documents`, `Downloads`, and so on for each user.

**`Windows`**:
    - The main directory containing the Windows operating system files.

**`hiberfil.sys`**:
    - A **system file** used by Windows to store the system state when it goes into **hibernation**. This file is often large and contains everything stored in memory (RAM).

In [ ]:
%%bash
pwd
cd /
ls -l

cd -  # go to previous directory

/g/My Drive/Upskilling/Github repository/Rep1/Linux_shell
total 5520
-rw-r--r-- 1 kavis 197609   18765 Sep 24 12:26 LICENSE.txt
-rw-r--r-- 1 kavis 197609  268097 Sep 24 12:26 ReleaseNotes.html
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 bin
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 cmd
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 dev
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 etc
-rwxr-xr-x 1 kavis 197609  139144 Sep 24 12:05 git-bash.exe
-rwxr-xr-x 1 kavis 197609  138616 Sep 24 12:05 git-cmd.exe
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 mingw64
dr-xr-xr-x 8 kavis 197609       0 Oct  6 16:00 proc
drwxr-xr-x 1 kavis 197609       0 Oct  6 15:59 tmp
-rw-r--r-- 1 kavis 197609 1315870 Oct  5 17:34 unins001.dat
-rwxr-xr-x 1 kavis 197609 3384040 Oct  5 17:34 unins001.exe
-rw-r--r-- 1 kavis 197609   24183 Oct  5 17:34 unins001.msg
drwxr-xr-x 1 kavis 197609       0 Oct  5 17:34 usr
/g/My Drive/Upskilling/Github repository/Rep1/Linux_shell


/bin -       essential command binaries - files in machine language for processor
/usr/bin -   User commands
/usr/lib -   Libraries

In [ ]:
%%bash
date
date -R

cal

cal sep 1991

ncal sept 2024

free


Sun Oct  6 12:44:01 UTC 2024
Sun, 06 Oct 2024 12:44:01 +0000
    October 2024      
Su Mo Tu We Th Fr Sa  
       1  2  3  4  5  
 6  7  8  9 10 11 12  
13 14 15 16 17 18 19  
20 21 22 23 24 25 26  
27 28 29 30 31        
                      
   September 1991     
Su Mo Tu We Th Fr Sa  
 1  2  3  4  5  6  7  
 8  9 10 11 12 13 14  
15 16 17 18 19 20 21  
22 23 24 25 26 27 28  
29 30                 
                      
    September 2024    
Su  1  8 15 22 29   
Mo  2  9 16 23 30   
Tu  3 10 17 24      
We  4 11 18 25      
Th  5 12 19 26      
Fr  6 13 20 27      
Sa  7 14 21 28      
               total        used        free      shared  buff/cache   available
Mem:         7993720     1643992     6086412        3100      535396     6349728
Swap:        2097152           0     2097152
